ruleguard/parser.go

```diff
@@ -676,21 +670,17 @@ func (p *rulesParser) toFilterOperand(e ast.Expr) filterOperand {
 		}
 		e = selector.X
 	}
+
 	indexing, ok := e.(*ast.IndexExpr)
-	if !ok {
-		return o
-	}
-	mapIdent, ok := indexing.X.(*ast.Ident)
-	if !ok {
-		return o
-	}
-	indexString, ok := p.toStringValue(indexing.Index)
-	if !ok {
-		return o
+	if ok {
+		mapIdent, ok := indexing.X.(*ast.Ident)
```

Comment:
`mapIdent` can be nil, isn't it?

Reply:
so the line below must be under next `if ok`

Reply:
Fixed.